src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs

```diff
@@ -4206,3 +4206,175 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
         info.AddValue("IntValue", IntProperty);
     }
 }
+
+[XmlInclude(typeof(DerivedClass))]
+public class BaseClass
+{
+    public string value { get; set; }
+    public string Value;
+}
+
+public class DerivedClass : BaseClass
+{
+    public new string value;
+    public new string Value { get; set; }
+}
+
+public class DefaultValuesSetToNaN
+{
+    [DefaultValue(double.NaN)]
+    public double DoubleProp { get; set; }
+
+    [DefaultValue(float.NaN)]
+    public float FloatProp { get; set; }
+
+    [DefaultValue(Double.NaN)]
+    public Double DoubleField;
+
+    [DefaultValue(Single.NaN)]
+    public Single SingleField;
+
+    public override bool Equals(object obj)
+    {
+        var other = obj as DefaultValuesSetToNaN;
+        return other == null ? false :
+            other.DoubleProp == this.DoubleProp && other.FloatProp == this.FloatProp &&
+            other.DoubleField == this.DoubleField && other.SingleField == this.SingleField;
+    }
+
+    public override int GetHashCode()
+    {
+        return this.DoubleProp.GetHashCode() ^ this.FloatProp.GetHashCode() ^
+            this.DoubleField.GetHashCode() ^ this.SingleField.GetHashCode();
+    }
+}
+
+[XmlRootAttribute("PurchaseOrder", Namespace = "http://www.cpandl.com", IsNullable = false)]
```

Comment:
In test method Xml_NullRefInXmlSerializerCtorTest(),  the defaultNamespace was set to  "http://www.contoso.com", can it be by design of the test to verify that the serialized xmlns is what's set in [XmlRootAttribute] instead of the default one specified in the XmlSerializer's constructor?